### PR TITLE
nixos/pam: add option failDelay

### PR DESCRIPTION
**Pull request by ocfox**
###### Description of changes
Add option failDelay for #186109

###### Things done

- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/security/pam.nix
Comment (NetaliDev) [inline nixos/modules/security/pam.nix:391]:
```suggestion
            If enabled, this will replace the `FAIL_DELAY' setting from <literal>login.defs</literal>.
```

File: nixos/modules/security/pam.nix
Comment (NetaliDev) [inline nixos/modules/security/pam.nix:400]:
```suggestion
          description = lib.mdDoc "The delay time (in microseconds) on failure.";
```

**Review by NetaliDev — CHANGES_REQUESTED** [review]
Except for a few typo issues, this looks good to me.

**Comment by NetaliDev** [comment]
I have tested it and the option does what it should do. So after the typo fixes, this PR should be ready to be merged.

File: nixos/modules/security/pam.nix
Comment (ocfox) [reply nixos/modules/security/pam.nix]: It looks like there was a little problem https://github.com/NixOS/nixpkgs/runs/7880374236?check_suite_focus=true#step:5:229.
`NotImplementedError: ('md node not supported yet', 'inline_html', ('<literal>',))`

File: nixos/modules/security/pam.nix
Comment (NetaliDev) [inline nixos/modules/security/pam.nix:391]:
```suggestion
            If enabled, this will replace the `FAIL_DELAY` setting from `login.defs`.
```
That should work now.

**Review by NetaliDev — CHANGES_REQUESTED** [review]

**Review by NetaliDev — APPROVED** [review]
Now it looks ready for me. I am not a committer so I can't merge this, but it is ready to merge in my opinion.

**Review by oluceps — APPROVED** [review]

**Comment by luochen1990** [comment]
I'm not familiar with these part, but will it be better for `failDelay.delay` to use `int` type instead of `str`, since it represents microseconds ?

**Comment by ocfox** [comment]
Thanks for the advice. I think so.

File: nixos/modules/security/pam.nix
Comment (bobby285271) [inline nixos/modules/security/pam.nix:556]: Did not test, shall we use `${toString cfg.failDelay.delay}`?

Comment (ocfox) [reply nixos/modules/security/pam.nix:556]: Thanks, I forget this.

**Comment by luochen1990** [comment]
LGTM

**Pull request merged**
